Javadoc errors

Tried to do a release but newly added javadocs are not building:
```java
[INFO] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.0.1:jar (attach-javadocs) on project webtau-core: MavenReportException: Error while generating Javadoc:
[INFO] [ERROR] Exit code: 1 - /Users/karol/OSSDev/webtau/target/checkout/webtau-core/src/main/java/com/twosigma/webtau/Matchers.java:31: error: bad use of '>'
[INFO] [ERROR]      * code(() -> {
[INFO] [ERROR]                 ^
[INFO] [ERROR] /Users/karol/OSSDev/webtau/target/checkout/webtau-core/src/main/java/com/twosigma/webtau/Matchers.java:143: error: bad use of '>'
[INFO] [ERROR]      * code(() -> {
[INFO] [ERROR]                 ^
[INFO] [ERROR] /Users/karol/OSSDev/webtau/target/checkout/webtau-core/src/main/java/com/twosigma/webtau/Matchers.java:159: error: bad use of '>'
[INFO] [ERROR]      * code(() -> {
[INFO] [ERROR]                 ^
[INFO] [ERROR] /Users/karol/OSSDev/webtau/target/checkout/webtau-core/src/main/java/com/twosigma/webtau/Matchers.java:175: error: bad use of '>'
[INFO] [ERROR]      * code(() -> {
[INFO] [ERROR]                 ^
[INFO] [ERROR] /Users/karol/OSSDev/webtau/target/checkout/webtau-core/src/main/java/com/twosigma/webtau/Matchers.java:191: error: bad use of '>'
[INFO] [ERROR]      * code(() -> {
[INFO] [ERROR]                 ^
[INFO] [ERROR] /Users/karol/OSSDev/webtau/target/checkout/webtau-core/src/main/java/com/twosigma/webtau/Matchers.java:208: error: bad use of '>'
[INFO] [ERROR]      * code(() -> {
[INFO] [ERROR]                 ^
[INFO] [ERROR] /Users/karol/OSSDev/webtau/target/checkout/webtau-core/src/main/java/com/twosigma/webtau/Matchers.java:231: warning: no @param for expected
[INFO] [ERROR]     public static GreaterThanMatcher beGreaterThan(Object expected) {
[INFO] [ERROR]                                      ^
[INFO] [ERROR] /Users/karol/OSSDev/webtau/target/checkout/webtau-core/src/main/java/com/twosigma/webtau/Matchers.java:231: warning: no @return
[INFO] [ERROR]     public static GreaterThanMatcher beGreaterThan(Object expected) {
[INFO] [ERROR]                                      ^
[INFO] [ERROR] /Users/karol/OSSDev/webtau/target/checkout/webtau-core/src/main/java/com/twosigma/webtau/Matchers.java:244: warning: no @param for expected
[INFO] [ERROR]     public static GreaterThanOrEqualMatcher beGreaterThanOrEqual(Object expected) {
[INFO] [ERROR]                                             ^
[INFO] [ERROR] /Users/karol/OSSDev/webtau/target/checkout/webtau-core/src/main/java/com/twosigma/webtau/Matchers.java:244: warning: no @return
[INFO] [ERROR]     public static GreaterThanOrEqualMatcher beGreaterThanOrEqual(Object expected) {
[INFO] [ERROR]                                             ^
[INFO] [ERROR] /Users/karol/OSSDev/webtau/target/checkout/webtau-core/src/main/java/com/twosigma/webtau/Matchers.java:257: warning: no @param for expected
[INFO] [ERROR]     public static LessThanMatcher beLessThan(Object expected) {
[INFO] [ERROR]                                   ^
[INFO] [ERROR] /Users/karol/OSSDev/webtau/target/checkout/webtau-core/src/main/java/com/twosigma/webtau/Matchers.java:257: warning: no @return
[INFO] [ERROR]     public static LessThanMatcher beLessThan(Object expected) {
[INFO] [ERROR]                                   ^
[INFO] [ERROR] /Users/karol/OSSDev/webtau/target/checkout/webtau-core/src/main/java/com/twosigma/webtau/Matchers.java:270: warning: no @param for expected
[INFO] [ERROR]     public static LessThanOrEqualMatcher beLessThanOrEqual(Object expected) {
[INFO] [ERROR]                                          ^
[INFO] [ERROR] /Users/karol/OSSDev/webtau/target/checkout/webtau-core/src/main/java/com/twosigma/webtau/Matchers.java:270: warning: no @return
[INFO] [ERROR]     public static LessThanOrEqualMatcher beLessThanOrEqual(Object expected) {
[INFO] [ERROR]                                          ^
[INFO] [ERROR] /Users/karol/OSSDev/webtau/target/checkout/webtau-core/src/main/java/com/twosigma/webtau/data/traceable/TraceableValue.java:65: warning: no @param for <R>
[INFO] [ERROR]     public static <R> R withDisabledChecks(Supplier<R> code) {
[INFO] [ERROR]                         ^
[INFO] [ERROR] /Users/karol/OSSDev/webtau/target/checkout/webtau-core/src/main/java/com/twosigma/webtau/data/traceable/TraceableValue.java:81: warning: no @param for <R>
[INFO] [ERROR]     public static <R> R withAlwaysFuzzyPassedChecks(Supplier<R> code) {
[INFO] [ERROR]                         ^
```

--- a/pom.xml
+++ b/pom.xml
@@ -601,20 +601,8 @@
                     <artifactId>maven-javadoc-plugin</artifactId>
                     <version>3.1.1</version>
                     <configuration>
-                        <javadocExecutable>${java.home}/bin/javadoc</javadocExecutable>
                         <detectOfflineLinks>false</detectOfflineLinks>
                     </configuration>
-                    <executions>
-                        <execution>
-                            <id>aggregate</id>
-                            <goals>
-                                <goal>aggregate</goal>
-                            </goals>
-                            <phase>site</phase>
-                            <configuration>
-                            </configuration>
-                        </execution>
-                    </executions>
                 </plugin>
             </plugins>
         </pluginManagement>
@@ -697,6 +685,13 @@
                         <goals>
                             <goal>jar</goal>
                         </goals>
+                    </execution>
+                    <execution>
+                        <id>aggregate</id>
+                        <goals>
+                            <goal>aggregate</goal>
+                        </goals>
+                        <phase>site</phase>
                     </execution>
                 </executions>
             </plugin>

--- a/pom.xml
+++ b/pom.xml
@@ -601,7 +601,8 @@
                     <artifactId>maven-javadoc-plugin</artifactId>
                     <version>3.1.1</version>
                     <configuration>
-                        <javadocExecutable>${java.home}/bin/javadoc</javadocExecutable>
+                        <javadocExecutable>${java.home}/../bin/javadoc</javadocExecutable>
+                        <detectOfflineLinks>false</detectOfflineLinks>
                     </configuration>
                     <executions>
                         <execution>
@@ -615,7 +616,6 @@
                         </execution>
                     </executions>
                 </plugin>
-
             </plugins>
         </pluginManagement>
 
@@ -686,6 +686,19 @@
                         </manifest>
                     </archive>
                 </configuration>
+            </plugin>
+
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-javadoc-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>attach-javadocs</id>
+                        <goals>
+                            <goal>jar</goal>
+                        </goals>
+                    </execution>
+                </executions>
             </plugin>
         </plugins>
     </build>
@@ -765,20 +778,6 @@
                                 <id>attach-sources</id>
                                 <goals>
                                     <goal>jar-no-fork</goal>
-                                </goals>
-                            </execution>
-                        </executions>
-                    </plugin>
-
-                    <plugin>
-                        <groupId>org.apache.maven.plugins</groupId>
-                        <artifactId>maven-javadoc-plugin</artifactId>
-                        <version>3.0.1</version>
-                        <executions>
-                            <execution>
-                                <id>attach-javadocs</id>
-                                <goals>
-                                    <goal>jar</goal>
                                 </goals>
                             </execution>
                         </executions>

--- a/pom.xml
+++ b/pom.xml
@@ -601,7 +601,7 @@
                     <artifactId>maven-javadoc-plugin</artifactId>
                     <version>3.1.1</version>
                     <configuration>
-                        <javadocExecutable>${java.home}/../bin/javadoc</javadocExecutable>
+                        <javadocExecutable>${java.home}/bin/javadoc</javadocExecutable>
                         <detectOfflineLinks>false</detectOfflineLinks>
                     </configuration>
                     <executions>

--- a/webtau-core/src/main/java/com/twosigma/webtau/Matchers.java
+++ b/webtau-core/src/main/java/com/twosigma/webtau/Matchers.java
@@ -28,7 +28,7 @@ public class Matchers {
     /**
      * Starting point of a code matcher
      * <pre>
-     * code(() -> {
+     * code(() -&gt; {
      *    businessLogic(-10);
      * }).should(throwException(IllegalArgumentException.class, "negatives are not allowed"));
      * </pre>
@@ -140,7 +140,7 @@ public class Matchers {
     /**
      * Throw exception <code>code</code> matcher.
      * <pre>
-     * code(() -> {
+     * code(() -&gt; {
      *     businessLogic(-10);
      * }).should(throwException("negatives are not allowed"));
      * </pre>
@@ -156,7 +156,7 @@ public class Matchers {
     /**
      * Throw exception <code>code</code> matcher.
      * <pre>
-     * code(() -> {
+     * code(() -&gt; {
      *      businessLogic(-10);
      * }).should(throwException(Pattern.compile("negative .* not allowed")));
      * </pre>
@@ -172,7 +172,7 @@ public class Matchers {
     /**
      * Throw exception <code>code</code> matcher.
      * <pre>
-     * code(() -> {
+     * code(() -&gt; {
      *      businessLogic(-10);
      * }).should(throwException(IllegalArgumentException.class));
      * </pre>
@@ -188,7 +188,7 @@ public class Matchers {
     /**
      * Throw exception <code>code</code> matcher.
      * <pre>
-     * code(() -> {
+     * code(() -&gt; {
      *      businessLogic(-10);
      * }).should(throwException(IllegalArgumentException.class, Pattern.compile("negative .* not allowed")));
      * </pre>
@@ -205,7 +205,7 @@ public class Matchers {
     /**
      * Throw exception <code>code</code> matcher.
      * <pre>
-     * code(() -> {
+     * code(() -&gt; {
      *      businessLogic(-10);
      * }).should(throwException(IllegalArgumentException.class, "negatives are not allowed"));
      * </pre>
@@ -225,6 +225,8 @@ public class Matchers {
      * <pre>
      * actual(value).shouldBe(greaterThan(10));
      * </pre>
+     * @param expected value to be greater than
+     * @return matcher instance
      * @see #greaterThan(Object)
      */
     @Deprecated
@@ -238,6 +240,8 @@ public class Matchers {
      * <pre>
      * actual(value).shouldBe(greaterThanOrEqual(10));
      * </pre>
+     * @param expected value to be greater than or equal
+     * @return matcher instance
      * @see #greaterThanOrEqual(Object)
      */
     @Deprecated
@@ -251,6 +255,8 @@ public class Matchers {
      * <pre>
      * actual(value).shouldBe(lessThan(10));
      * </pre>
+     * @param expected value to be less than
+     * @return matcher instance
      * @see #lessThan(Object)
      */
     @Deprecated
@@ -264,6 +270,8 @@ public class Matchers {
      * <pre>
      * actual(value).shouldBe(lessThanOrEqual(10));
      * </pre>
+     * @param expected value to be less than
+     * @return matcher instance
      * @see #lessThanOrEqual(Object)
      */
     @Deprecated

--- a/webtau-core/src/main/java/com/twosigma/webtau/data/traceable/TraceableValue.java
+++ b/webtau-core/src/main/java/com/twosigma/webtau/data/traceable/TraceableValue.java
@@ -60,6 +60,7 @@ public class TraceableValue {
      * (e.g. using for just comparison)
      *
      * @param code code to execute
+     * @param <R> type of the returned value
      * @return value returned by the passed code
      */
     public static <R> R withDisabledChecks(Supplier<R> code) {
@@ -76,6 +77,7 @@ public class TraceableValue {
      * (e.g. using nodes inside if-else like constructs)
      *
      * @param code code to execute
+     * @param <R> type of the returned value
      * @return value returned by the passed code
      */
     public static <R> R withAlwaysFuzzyPassedChecks(Supplier<R> code) {

